src/extension.c

```diff
@@ -56,20 +56,33 @@ enum ExtensionState
 };
 
 static enum ExtensionState extstate = EXTENSION_STATE_INITIAL;
-static TransactionId drop_transaction_id = InvalidTransactionId;
 
 bool
 extension_is_being_dropped(Oid relid)
 {
+	/*
+	 * during a extension drop, the extension proxy table will be dropped.
+	 * This will automatically cause a relcache invalidation on the proxy
+	 * tables's OID.
+	 */
 	return relid == extension_proxy_oid;
 }
 
+
+static bool
+proxy_table_exists()
```

Comment:
Is this the best indicator we have of "being in the process of dropping the extension"? I am asking, because the `drop_transaction_id` check from before was at least explicit, rather than this check that depends on the implicit ordering of removal events and the presence of certain tables.

One reason I am asking is because we might decide in the future that we do not need this proxy table, then this check needs to be changed to something else. Another reason is that the ordering of events might not always be the same, although I agree that it is unlikely that the proxy table is removed after the extension. But, on the other hand, the internal semantics of how this happens might change and this check risks breaking.

Wouldn't it be better to just patch up the `drop_transaction_id` mechanism to also check for other DDL DROP commands that can cascade to dropping the extension? For example, we could check for DROP schema and check whether it is the extension's schema. This is a bit more code, but is at least explicit.